csharp/examples/IoTDataPipelineExample/Model/SensorDataPipeline.cs

```diff
@@ -0,0 +1,225 @@
+                {"catch", "O"},
+                {"dribbling", "P"},
+                {"writing", "Q"},
+                {"clapping", "R"},
+                {"folding", "S"},
+            };
+
+        public SensorDataPipeline(int totalInputs, int queueCapacity)
+        {
+            _totalInputs = totalInputs;
+            _queueCapacity = queueCapacity;
+            _channel = Channel.CreateBounded<SensorData>(_queueCapacity);
+            _writer = _channel.Writer;
+            _reader = _channel.Reader;
+            _colSubjectIdBuilderDict = new Dictionary<int, Int32Array.Builder>();
+            _colActivityLabelBuilderDict = new Dictionary<int, StringArray.Builder>();
+            _colTimestampBuilderDict = new Dictionary<int, TimestampArray.Builder>();
+            _colXAxisBuilderDict = new Dictionary<int, DoubleArray.Builder>();
+            _colYAxisBuilderDict = new Dictionary<int, DoubleArray.Builder>();
+            _colZAxisBuilderDict = new Dictionary<int, DoubleArray.Builder>();
+        }
+
+        public async Task WriteToChannel(int taskNumber)
+        {
+            Random rand = new Random();
+            List<string> keyList = new List<string>(activityLabel.Keys);
+            int count = keyList.Count;
+            var basis = DateTimeOffset.UtcNow;
+
+            Console.WriteLine($"Write to channel task {taskNumber} started!");
+            while (await _writer.WaitToWriteAsync())
+            {
+                string randomKey = keyList[rand.Next(count)];
+                string label = activityLabel[randomKey];
+
+                // generate random missing values
+                if (rand.Next(10_000) == 9_999)
+                {
+                    label = null;
+                }
+
+                var item = new SensorData
+                {
+                    // approximately 9_000 unique subjects/sensors
+                    subjectId = rand.Next(1_001, 1_123),
+                    activityLabel = label,
+                    timestamp = basis.AddMilliseconds(1),
+                    x_Axis = rand.NextDouble(),
+                    y_Axis = rand.NextDouble(),
+                    z_Axis = rand.NextDouble(),
+                };
+
+                if (_writer.TryWrite(item))
+                {
+                    Interlocked.Increment(ref _size);
+
+                    if (_size >= _totalInputs)
+                    {
+                        _writer.TryComplete();
+                    }
+                }
+            }
+
+            Console.WriteLine($"Write to channel task {taskNumber} finished!");
+        }
+
+        public async Task ReadFromChannel()
+        {
+
+            Console.WriteLine($"Read from channel task started!");
+            while (await _reader.WaitToReadAsync())
+            {
+                while (_reader.TryRead(out SensorData item))
+                {
+                    int subjectId = (int)item.subjectId;
+
+                    if (item != null)
+                    {
```

Comment:
We should be checking for `null` before dereferencing `item`.
```suggestion
                    if (item != null)
                    {
                        int subjectId = (int)item.subjectId;
```